src/librbd/io/ReadResult.h

```diff
@@ -38,15 +38,13 @@ class ReadResult {
 
   struct C_ObjectReadRequest : public Context {
     AioCompletion *aio_completion;
-    uint64_t object_off;
-    uint64_t object_len;
+    const Extents object_extents;
     LightweightBufferExtents buffer_extents;
 
     bufferlist bl;
     Extents extent_map;
 
-    C_ObjectReadRequest(AioCompletion *aio_completion, uint64_t object_off,
-                        uint64_t object_len,
+    C_ObjectReadRequest(AioCompletion *aio_completion, const Extents &extents,
```

Comment:
Nit: let's use an rvalue reference so we can use move semantics
```suggestion
    C_ObjectReadRequest(AioCompletion *aio_completion, Extents&& extents,
```